pyerrors/input/json.py

```diff
@@ -0,0 +1,332 @@
+            d['reweighted'] = ol[0].reweighted
+        d['value'] = [o.value for o in ol]
+        d['data'] = _gen_data_d_from_list(ol)
+
+        return d
+
+    def write_Array_to_dict(oa):
+        ol = np.ravel(oa)
+        _assert_equal_properties(ol)
+        d = {}
+        d['type'] = 'Array'
+        d['layout'] = str(oa.shape).lstrip('(').rstrip(')')
+        if ol[0].tag:
+            d['tag'] = ol[0].tag
+            if isinstance(ol[0].tag, str):
+                if len(set([o.tag for o in ol])) > 1:
+                    d['tag'] = ''
+                    for o in ol:
+                        d['tag'] += '%s\n' % (o.tag)
+        if ol[0].reweighted:
+            d['reweighted'] = ol[0].reweighted
+        d['value'] = [o.value for o in ol]
+        d['data'] = _gen_data_d_from_list(ol)
+        return d
+    if not isinstance(ol, list):
+        ol = [ol]
+    d = {}
+    d['program'] = 'pyerrors %s' % (pyerrorsversion.__version__)
+    d['version'] = '0.1'
+    d['who'] = getpass.getuser()
+    d['date'] = str(datetime.datetime.now())[:-7]
+    d['host'] = socket.gethostname() + ', ' + platform.platform()
+
+    if description:
+        d['description'] = description
+    d['obsdata'] = []
+    for io in ol:
+        if isinstance(io, Obs):
+            d['obsdata'].append(write_Obs_to_dict(io))
+        elif isinstance(io, list):
+            d['obsdata'].append(write_List_to_dict(io))
+        elif isinstance(io, np.ndarray):
+            d['obsdata'].append(write_Array_to_dict(io))
+
+    jsonstring = json.dumps(d, indent=indent, cls=my_encoder)
+    # workaround for un-indentation of delta lists
+    jsonstring = jsonstring.replace('    "[', '    [').replace(']",', '],').replace(']"\n', ']\n')
+
+    return jsonstring
+
+
+def dump_to_json(ol, fname, description='', indent=1, gz=True):
+    """Export a list of Obs or structures containing Obs to a .json(.gz) file
+
+    Parameters
+    -----------------
+    ol : list
+        List of objects that will be exported. At the moments, these objects can be
+        either of: Obs, list, np.ndarray
+        All Obs inside a structure have to be defined on the same set of configurations.
+    fname : str
+        Filename of the output file
+    description : str
+        Optional string that describes the contents of the json file
+    indent : int
+        Specify the indentation level of the json file. None or 0 is permissible and
+        saves disk space.
+    gz : bool
+        If True, the output is a gzipped json. If False, the output is a json file.
+    """
+
+    jsonstring = create_json_string(ol, fname, description, indent)
+
+    if not fname.endswith('.json') and not fname.endswith('.gz'):
+        fname += '.json'
+
+    if gz:
+        if not fname.endswith('.gz'):
+            fname += '.gz'
+
+        fp = gzip.open(fname, 'wb')
+        fp.write(jsonstring.encode('utf-8'))
+    else:
+        fp = open(fname, 'w')
+        fp.write(jsonstring)
```

Comment:
I'm not sure. This could be done via 
fp = open(fname, 'w', encoding='utf-8')
but it's no fun to read the resulting *.json file using less because of all the additional printout. However, JSON encoders are required to be able to interpret utf-8 and therefore I/O would still work as expected. I do not intend to use plain json, anyways.